pkg/metrics/metric/metric.go

```diff
@@ -196,3 +239,72 @@ func (b Opts) GetConfigName() string {
 	}
 	return b.ConfigName
 }
+
+type Label struct {
```

Comment:
Might be a good idea to add tests for some of the calls below...

Reply:
good idea